waku/v2/node/rest/store/handlers.nim

```diff
@@ -0,0 +1,251 @@
+when (NimMajor, NimMinor) < (1, 4):
+  {.push raises: [Defect].}
+else:
+  {.push raises: [].}
+
+import
+  stew/results,
+  chronicles,
+  json_serialization,
+  presto/route
+import
+  ../../../../common/base64,
+  ../../../protocol/waku_message/topics/content_topic,
+  ../../../protocol/waku_store/common,
+  ../../../utils/time,
+  ../../waku_node,
+  ../../peer_manager,
+  ../responses,
+  ../serdes,
+  ./types
+
+export types
+
+logScope:
+  topics = "waku node rest store_api"
+
+const futTimeout* = 5.seconds # Max time to wait for futures
+
+# Queries the store-node with the query parameters and
+# returns a RestApiResponse that is sent back to the api client.
+proc makeHistoryQuery(selfNode: WakuNode,
```

Comment:
Nitpick, but perhaps something like `performHistoryQuery` is a better proc name here? To me, `make` sounds similar to `create`. :)